crates/router/src/core/utils.rs

```diff
@@ -1377,6 +1377,13 @@ impl<T, F> GetProfileId for (storage::Payouts, T, F) {
     }
 }
 
+#[cfg(feature = "recon")]
+impl GetProfileId for storage::UserRole {
+    fn get_profile_id(&self) -> Option<&String> {
+        self.profile_id.as_ref()
+    }
+}
```

Comment:
This validation is not sufficient to say whether a profile_id is part of a specific user_role.

User role can be at merchant level or org level in which case you will have to check whether the merchant_id or org_id in the user_role is parent of the profile asked.